### ISSUE TITLE
Explain authorization workflow on login page

### ISSUE THREAD

**Issue by jeepfromoil** [Vavassor/Tusky#69]
Pressing the `login with mastodon` on the login page opens the URL of the instance login page of via the Web browser.  Neither Tusky nor the instance Web page explains why it's required to log into Mastodon on the Web to connect the app to the user account.

A few words on these login pages could explain the process:

- on Tusky: add *Log into your instance on the Web to allow Tusky to connect to your account*
- on Mastodon: add *Authorize <app_name> to connect to your <instance_name> account*

At first, I thought I was directed to the Web because Tusky was down or because the app wasn't compatible with my instance.

**Comment by STPKITT** [comment]
I second this! I can't even login via Tusky because I always get the error screen with the angry elephant and the error message "Security verification failed. Are you blocking cookies?". I do not even know though which browser (I've got several installed) it uses.
Is Tusky just a wrapper of a website or a native app?

**Comment by connyduck** [comment]
Yes the login screen definitely needs some improvement 🤔 It might require a total redesign I am not sure if slapping some text on it solves the problems it has.

> I can't even login via Tusky because I always get the error screen with the angry elephant and the error message "Security verification failed. Are you blocking cookies?". I do not even know though which browser (I've got several installed) Tusky uses for the login.

It uses your default browser, and the error you are describing can usually be resolved by using another browser. If not, please tell me your instance, your browser and your Android version so I can investigate further.
> Is Tusky just a wrapper of a website or a native app?

a native app

**Comment by STPKITT** [comment]
Thanks for you reply!

I do not have a default browser defined, but in Tusky the popup that usually asks which browser to use never popped up. I just found out though by pressing the 3-dot-menu while being on the login page it says "provided by Fennec" or something similar (I'm on a German system). In that menu I could click "open in browser" and only then Android asked me which browser to use. I could then, after adjusting cookie settings of Fennec, successfully log in :-)

If Tusky is a native app I wonder why it uses a webbrowser for the login process. I also know Mastalab which doesn't use the browser for login as far as I can tell and therefor I never ran into a cookie related problem.

Sorry that it got a bit offtopic here, but my experience still may help to identify the hurdles a user might face in the authorization workflow.

**Comment by charlag** [comment]
@STPKITT thanks for the elaborate explanations!
We use browser for the login process because:
1) OAuth is quite a complex beast. When there's a two-factor auth it becomes even more complex. We would like to have another flow but it's a hard task and we don't have enough time now to implement it
2) It is sometimes more convenient to just press a button in the browser than remember you password and type it in

In short: we know it's a pain point but it's not top priority now.

**Comment by STPKITT** [comment]
Ok, I understand that very well.

What could be implemented easily though I think and already help a bit would be to trigger the Android popup, which asks the user which browser to use, earlier in the process when the user clicks on "Login with Mastodon". Maybe that popup is even possible to trigger for users who have defined a default browser.

**Comment by dioraman** [comment]
It is unclear to me too. I second this.
